### ISSUE TITLE
install & check does not work

### ISSUE THREAD

**Issue by ezradlesser** [jetflav/CMPPlugin#1]
`make install` and `make check` do not work since there is no utils directory on master.

**Comment by RenePoncelet** [comment]
I tried to reproduce your problem, but couldn't. The ./utils directory is part of the fjcontrib package. So if you placed the directory in fjcontrib and go inside of CMPPlugin and execute `make install` it should work (at least for me it does).

**Comment by ezradlesser** [comment]
Hi Rene, thanks for the reply. I wasn't aware that ./utils was inside fjcontrib... two of the other three jetflav repos have it as part of their own code. I guess it would be good to standardize (or write a README.md with the install directions, if different from the others). Thanks again!

**Comment by RenePoncelet** [comment]
Ah, I see. Not sure what the default is from fjcontrib side. Looking into some of the older contribs it seems that they also don't have the utils directory inside. @scybozl : do you know what the correct conventions are?

**Comment by gavinsalam** [comment]
(Adding at least one member from other flavour algs)

What we should really do is to get all the flavour plugins into fjcontrib proper. That didn’t make sense when things were still potentially in flux, but I get the impression things are relatively stable.
The constraints for fjcontrib are

1) things should be moved into the fjcontrib svn area (we know this is prehistoric, but haven’t had time to revamp the scripts to work with git; probably not too hard but someone would need to spend a day on it)

2) fjcontrib doesn’t allow dependencies between contribs, so CMP, IFN and GHS should probably go into a single contrib, since they share the flavour infrastructure. SDF is more complex because it depends on SoftDrop. I don’t know how deep that dependency is or whether it’s just a few lines that could be coded up separately.

Best
Gavin

On 12 Mar 2024, at 12:59, RenePoncelet ***@***.***> wrote:

﻿

Ah, I see. Not sure what the default is from fjcontrib side. Looking into some of the older contribs it seems that they also don't have the utils directory inside. @scybozl<https://github.com/scybozl> : do you know what the correct conventions are?

**Comment by scybozl** [comment]
I agree it would be nice to have all the algorithms in fjcontrib (probably also for experimentalists as they may have a well-oiled procedure for installing additional contribs!). My take:

1) fjcontrib being on svn is not a show-stopper, even though it's not ideal
2) and I think it's fine to have all of the algos in the same contrib. Is a mechanism to handle inter-contrib dependencies in fjcontrib something that could be done easily? I guess it's not too convoluted to copy bits of code from SoftDrop but it does not look like just a couple of lines (I'll let the SDF authors correct me here)

**Comment by gavinsalam** [comment]
for the inter-contrib dependencies, it’s not totally trivial — in a sense it’s currently a design "feature”, in that it allows each contrib to be worked on independently of the rest. In practice it’s a question of

1) making sure that the Makefile understands dependencies between directories, in a way that can be set up as part of the configure step, which writes the top-level Makefile. I’m not entirely sure how to do that, but I guess one could figure it out.

2) deciding whether we need to encode versioning constraints between contribs in some way. (This is secondary)





On 13 Mar 2024, at 01:20, Ludovic Scyboz ***@***.***> wrote:



I agree it would be nice to have all the algorithms in fjcontrib (probably also for experimentalists as they may have a well-oiled procedure for installing additional contribs!). My take:

  1.  fjcontrib being on svn is not a show-stopper, even though it's not ideal
  2.  and I think it's fine to have all of the algos in the same contrib. Is a mechanism to handle inter-contrib dependencies in fjcontrib something that could be done easily? I guess it's not too convoluted to copy bits of code from SoftDrop but it does not look like just a couple of lines (I'll let the SDF authors correct me here)

**Comment by gavinsalam** [comment]
Hi Gavin, everyone,

I’m adding Daniel and Simone C to this thread. They’re happy to follow up on this for SDF.

Cheers,
Simone
-----------
Simone Marzani  - Associate Professor
Dipartimento di Fisica, Università di Genova, and
INFN - Sezione di Genova.
Via Dodecaneso 33, 16146 Genova Italy.
Phone: +39 010 353 6212.
https://www.ge.infn.it/~smarzani 

** My working hours may not be your working hours. Please do not feel obliged to reply outside of your normal work schedule **

**Comment by gavinsalam** [comment]
Hi Gavin,

I should have time in the next days to play with the contribs. However, I admit I had to google what SVN is since I never used it.
Do you need to give us access the the repo? How do we want to coordinate since all the algos should be part of the same contrib?

Best,
Simone C

13 Mar 2024 07:26:47 Simone Marzani ***@***.***>:

> Hi Gavin, everyone,
> 
> I’m adding Daniel and Simone C to this thread. They’re happy to follow up on this for SDF.
> 
> Cheers,
> Simone
> -----------
> Simone Marzani  - Associate Professor
> Dipartimento di Fisica, Università di Genova, and
> INFN - Sezione di Genova.
> Via Dodecaneso 33, 16146 Genova Italy.
> Phone: +39 010 353 6212.
> https://www.ge.infn.it/~smarzani 
> 
> ** My working hours may not be your working hours. Please do not feel obliged to reply outside of your normal work schedule **
> 
> 
> 
> 
>

**Comment by ezradlesser** [comment]
From the experimentalist side, I also support making a working fjcontrib contribution with all the algorithms included here. That should resolve many of the issues that I had when trying to build, check, install, and use these plugins (e.g. copying `utils/` from one contribution to another, renaming wrongly-named files, making `*.so` dynamic libraries to load with `cppyy` in python...).

Nonetheless, I believe that this issue should still be understood first:
https://github.com/jetflav/GHSAlgo/issues/1
as (for me) it seems to indicate that either there was, or currently there is, a bug somewhere in that particular algorithm...
